lib/ruumba/analyzer.rb

```diff
@@ -61,12 +61,50 @@ def copy_erb_files(fq_dir, tmp, pwd)
       end
     end
 
+    def autofix!(target, tmp)
+      src_dir = Pathname.new(File.expand_path(target))
+
+      Dir.foreach(tmp) do |f|
+        # Match the extracted code from the tmp directory
+        # to its original file in the source directory in
+        # order to find-and-replace for --auto-correct.
+        src = f.split('.').tap(&:pop).join '.'
+
+        if File.extname(src) == '.erb' && File.exist?("#{src_dir}/#{src}")
+          # Walk through the extracted .erb.rb file and write
+          # its contents into the interpolation areas of the
+          # original .erb file.
+          code       = File.readlines("#{tmp}/#{f}")
+          noncode    = File.readlines("#{src_dir}/#{src}")
+                           .reject { |l| l =~ ERB_REGEX }
+          to_correct = File.readlines("#{src_dir}/#{src}")
+          out        = []
+
+          to_correct.each do |line|
+            # @TODO: This is pretty hacky. (EQW 30 Aug 2017)
+            out << if line.match?(/<%/)
+                     "  <% #{code.shift.rstrip} %>\n"
+                   elsif line.match?(/<%=/)
+                     "  <%= #{code.shift.rstrip} %>\n"
+                   else
+                     noncode.shift
+                   end
+          end
+
+          File.open("#{src_dir}/#{src}", 'w') do |contents|
+            contents << out.join
+          end
+        end
+      end
+    end
+
     def run_rubocop(target, tmp)
       args = (@options[:arguments] || []).join(' ')
       todo = tmp + '.rubocop_todo.yml'
 
       system("cd #{tmp} && rubocop #{args} #{target}").tap do
         FileUtils.cp(todo, ENV['PWD']) if File.exist?(todo)
+        autofix!(target, tmp) if @options[:arguments]&.include? '--auto-correct'
```

Comment:
This is not compatible with old versions of ruby :(

Could be this implemented without the `&.` operator?

Reply:
Sure, I can downgrade this branch back to 2.2.2+ rather than 2.3.0+